### PR TITLE
[POC] pkg/validate: external validation runner

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
- pkg/validate: external validation runner

**Motivation for the change:** by running the following, you can try out this POC:

```sh
cat <<EOF > validator.sh
#!/bin/bash
echo '{"passed": false, "outputs": [{"type": "error", "message": "failed validator"}]}'
exit 0
EOF
chmod +x validator.sh
export OPERATOR_SDK_VALIDATOR_ENTRYPOINTS=$(pwd)/validator.sh
operator-sdk bundle validate ./bundle/ --output json-alpha1
```

You should see the following:

```console
$ operator-sdk bundle validate ./bundle/ --output json-alpha1
INFO[0000] Found annotations file                        bundle-dir=./bundle container-tool=docker
INFO[0000] Could not find optional dependencies file     bundle-dir=./bundle container-tool=docker
{
    "passed": false,
    "outputs": [
        {
            "type": "error",
            "message": "failed validator"
        }
    ]
}
$ echo $?
1
```

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by estroz** [comment]
I plan on writing an EP if this gets positive feedback.

File: pkg/validate/external.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+// GetExternalValidatorEntrypoints returns a list of external validator entrypoints
+// retrieved from ValidatorEntrypointsEnv and true if set. If not set or set to the empty string,
+// GetExternalValidatorEntrypoints returns false.
+func GetExternalValidatorEntrypoints() ([]string, bool) {
+	entrypoints, isSet := os.LookupEnv(ValidatorEntrypointsEnv)
+	if !isSet || strings.TrimSpace(entrypoints) == "" {
+		return nil, false
+	}
+	return filepath.SplitList(entrypoints), true
+}
```
Comment (jchunkins) [inline pkg/validate/external.go:49]: You might want to do some additional sanity checks on these values being provided.

1. Does the path being provided correlate to something on the file system?
1. Is this a directory?
2. Is this really an executable file?

Comment (jchunkins) [reply pkg/validate/external.go:49]: If validation of the entry point paths fails, probably makes sense to log the issue and strip that one out of the returned list so the rest of the system won't act on garbage entry points.

Comment (estroz) [reply pkg/validate/external.go:49]: `exec.Cmd.Run()` will return a fairly indicative error if a file cannot be found by path, is a directory, or if the file is not executable.

**Comment by asmacdo** [comment]
@estroz Since this has gotten pretty stale, I'm going to close this for now. If we need to keep moving with this one, lets create an issue or reopen the PR.

**Pull request closed without merging**